templates/python/.github/workflows/python-poetry-release.yaml

```diff
@@ -0,0 +1,57 @@
+name: "Release Python packages"
+description: "Release Python packages using Poetry to TestPyPI or PyPI"
+
+inputs:
+  pypi-token:
+    description: "The PyPI token for publishing packages."
+    required: true
+  publish-to-test:
+    description: "If set to false, the packages are published to PyPI. (Default is true)"
+    required: false
+    default: "true"
+  python-version:
+    description: "The Python version for the Poetry virtual environment. (Default is 3.10)"
+    required: false
+    default: "3.10"
+  poetry-version:
+    description: "The Poetry version to be installed. (Default is 1.2.2)"
+    required: false
+    default: "1.2.2"
+  working-directory:
+    description: "The working directory of your Python packages. (Default is root directory)"
+    required: false
+    default: "./"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set up Poetry ${{ inputs.poetry-version }}
```

Comment:
you could replace all this with a call to our reusable workflow https://github.com/bakdata/ci-templates/tree/main/.github/workflows#calling-the-workflow-3

```yaml
jobs:
  release:
    uses: bakdata/ci-templates/.github/workflows/python-poetry-release.yaml@main
    with:
      ...
```